lib/presentation/resources/todo_cards/todo_cards.screen.dart

```diff
@@ -0,0 +1,179 @@
+import 'package:clean_architecture_demo/presentation/resources/todo_cards/todo_crads_cubit.dart';
+import 'package:clean_architecture_demo/presentation/resources/values_manager.dart';
+import 'package:clean_architecture_demo/presentation/utils/ui_utils.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class TodoCardsSecreen extends StatefulWidget {
+  const TodoCardsSecreen({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _TodoCardsSecreenState();
+}
+
+class _TodoCardsSecreenState extends State<TodoCardsSecreen> {
+  late TodoCardsCubit _todoCardsCubit;
+
+  @override
+  void initState() {
+    super.initState();
+    initialise();
+  }
+
+  Future<void> initialise() async {
+    if (mounted) {
+      _todoCardsCubit =
+          TodoCardsCubit(todoRepository: getItInstance.get<TodoRepository>());
+
+      _todoCardsCubit.getTodoCards();
+    }
+  }
+
+  @override
+  void didChangeDependencies() {
+    //TODO: Initialize the localisation here
+    super.didChangeDependencies();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          //TODO: add from localisation
+          title: const Text('My TODOs'),
+          actions: [
+            _buildCalendarAppBarButton(),
+          ],
+        ),
+        body: BlocProvider(
+            create: (context) => _todoCardsCubit,
+            child: BlocBuilder<TodoCardsCubit, TodoCardsState>(
+                builder: (context, state) {
+              switch (state.runtimeType) {
+                case GetTodoCardsInProgress:
+                  return const Center(
+                    child: CircularProgressIndicator(),
+                  );
+                case GetTodoCardsSuccess:
+                  final todoCards = (state as GetTodoCardsSuccess).todoCards;
+                  return _buildBody(todoCards ?? []);
+                case GetTodoCardsFailure:
+                  return const Center(
+                    child: Column(
+                      children: [
+                        Icon(Icons.error, size: 50, color: Colors.red),
+                        //TODO: add from localisation
+                        Text('Failed to fetch todo cards'),
+                      ],
+                    ),
+                  );
+              }
+
+              return Container();
+            })),
+        floatingActionButton: FloatingActionButton(
+            onPressed: () {
+              //TODO: build the navigation to next screen
+            },
+            child: const Icon(Icons.add)));
+  }
+
+  Widget _buildCalendarAppBarButton() {
+    return Padding(
+      padding: const EdgeInsets.all(AppPadding.p20),
+      child: InkWell(
+        onTap: () => {
+          //TODO: navigate to calendar screen
+          print('Calendar button tapped'),
```

Comment:
@dilumdesilva for debug level events we can create a constants class called DebugLogs for ex and add all these constants in those files.

Reply:
This was just a print line for myself. It's removed.